src/disco/topo/fd_topo_run.c

```diff
@@ -0,0 +1,280 @@
+  fd_topo_run_tile( args.topo, args.tile, 0, args.uid, args.gid, -1, NULL, NULL, &args.tile_run, args.tile_align, args.tile_footprint );
+  FD_LOG_ERR(( "fd_topo_run_tile() returned" ));
+  return NULL;
+}
+
+static inline pthread_t
+run_tile_thread( fd_topo_t *         topo,
+                 fd_topo_tile_t *    tile,
+                 fd_topo_run_tile_t  tile_run,
+                 ulong (* tile_align )( fd_topo_tile_t const * tile ),
+                 ulong (* tile_footprint )( fd_topo_tile_t const * tile ),
+                 uint                uid,
+                 uint                gid,
+                 fd_cpuset_t const * floating_cpu_set,
+                 int                 floating_priority ) {
+  void * stack = fd_tile_private_stack_new( 1, tile->cpu_idx );
+
+  pthread_attr_t attr[ 1 ];
+  if( FD_UNLIKELY( pthread_attr_init( attr ) ) ) FD_LOG_ERR(( "pthread_attr_init() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
+  if( FD_UNLIKELY( pthread_attr_setstack( attr, stack, FD_TILE_PRIVATE_STACK_SZ ) ) ) FD_LOG_ERR(( "pthread_attr_setstacksize() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
+
+  FD_CPUSET_DECL( cpu_set );
+  if( FD_LIKELY( tile->cpu_idx<65535UL ) ) {
+    /* set the thread affinity before we clone the new process to ensure
+       kernel first touch happens on the desired thread. */
+    fd_cpuset_insert( cpu_set, tile->cpu_idx );
+    if( FD_UNLIKELY( -1==setpriority( PRIO_PROCESS, 0, -19 ) ) ) FD_LOG_ERR(( "setpriority() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
+  } else {
+    fd_memcpy( cpu_set, floating_cpu_set, fd_cpuset_footprint() );
+    if( FD_UNLIKELY( -1==setpriority( PRIO_PROCESS, 0, floating_priority ) ) ) FD_LOG_ERR(( "setpriority() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
+  }
+
+  if( FD_UNLIKELY( fd_cpuset_setaffinity( 0, cpu_set ) ) ) FD_LOG_ERR(( "sched_setaffinity failed (%i-%s)", errno, fd_io_strerror( errno ) ));
+
+  fd_topo_run_thread_args_t args = {
+    .topo           = topo,
+    .tile           = tile,
+    .tile_run       = tile_run,
+    .tile_align     = tile_align,
+    .tile_footprint = tile_footprint,
+    .uid            = uid,
+    .gid            = gid,
+    .copied         = 0,
+  };
+
+  pthread_t pthread;
+  if( FD_UNLIKELY( pthread_create( &pthread, attr, run_tile_thread_main, &args ) ) ) FD_LOG_ERR(( "pthread_create() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
+
+  while( !FD_VOLATILE( args.copied ) ) FD_SPIN_PAUSE();
+  return pthread;
+}
+
+void
+fd_topo_run_single_process( fd_topo_t * topo,
+                            int         solana_labs,
+                            uint        uid,
+                            uint        gid,
+                            fd_topo_run_tile_t (* tile_run )( fd_topo_tile_t * tile ),
+                            ulong (* tile_align )( fd_topo_tile_t const * tile ),
+                            ulong (* tile_footprint )( fd_topo_tile_t const * tile ) ) {
```

Comment:
I'm not sure if we have a convention for aligning function pointers, but I'd probably align them a bit better (and elsewhere)